<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab tab_name="RPY" containers="1">
   <Container>
    <DockSplitter sizes="0.333037;0.333927;0.333037" count="3" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="1.000000" bottom="0.000000" top="1.000000"/>
       <limitY/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="1.000000" bottom="0.000000" top="1.000000"/>
       <limitY/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="1.000000" bottom="0.000000" top="1.000000"/>
       <limitY/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="traj" containers="1">
   <Container>
    <DockSplitter sizes="1" count="1" orientation="-">
     <DockArea name="...">
      <plot mode="XYPlot" style="Lines" flip_x="false" flip_y="false">
       <range left="-0.097444" right="0.668537" bottom="-0.298157" top="0.442029"/>
       <limitY/>
       <curve curve_x="/quadrotor/pose/pose/pose/position/x" color="#ff7f0e" curve_y="/quadrotor/pose/pose/pose/position/y" name="/quadrotor/pose/pose/pose/position/[x;y]"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="xyz" containers="1">
   <Container>
    <DockSplitter sizes="0.333037;0.333927;0.333037" count="3" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-0.097444" top="0.668537"/>
       <limitY/>
       <curve color="#f14cc1" name="/quadrotor/pose/pose/pose/position/x"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-0.298157" top="0.442029"/>
       <limitY/>
       <curve color="#9467bd" name="/quadrotor/pose/pose/pose/position/y"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-1.363764" top="-1.185979"/>
       <limitY/>
       <curve color="#17becf" name="/quadrotor/pose/pose/pose/position/z"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="pqr" containers="1">
   <Container>
    <DockSplitter sizes="0.333037;0.333927;0.333037" count="3" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-6.371160" top="7.803941"/>
       <limitY/>
       <curve color="#bcbd22" name="/quadrotor/imu/angular_velocity/x"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-7.387050" top="7.935661"/>
       <limitY/>
       <curve color="#1f77b4" name="/quadrotor/imu/angular_velocity/y"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-4.677416" top="4.398317"/>
       <limitY/>
       <curve color="#d62728" name="/quadrotor/imu/angular_velocity/z"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="acc" containers="1">
   <Container>
    <DockSplitter sizes="0.333037;0.333927;0.333037" count="3" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-7.256142" top="7.394958"/>
       <limitY/>
       <curve color="#1ac938" name="/quadrotor/imu/linear_acceleration/x"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-5.794383" top="6.368795"/>
       <limitY/>
       <curve color="#ff7f0e" name="/quadrotor/imu/linear_acceleration/y"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-25.864225" top="0.527861"/>
       <limitY/>
       <curve color="#f14cc1" name="/quadrotor/imu/linear_acceleration/z"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="I_xyz" containers="1">
   <Container>
    <DockSplitter sizes="0.250446;0.249554;0.250446;0.249554" count="4" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="0.000513" top="0.001150"/>
       <limitY/>
       <curve color="#9467bd" name="/quadrotor/geom_inertia_estimator/param_estimates/I/x"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="0.000552" top="0.001160"/>
       <limitY/>
       <curve color="#17becf" name="/quadrotor/geom_inertia_estimator/param_estimates/I/y"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="0.000627" top="0.001112"/>
       <limitY/>
       <curve color="#bcbd22" name="/quadrotor/geom_inertia_estimator/param_estimates/I/z"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="0.229275" top="0.501803"/>
       <limitY/>
       <curve color="#1f77b4" name="/quadrotor/geom_inertia_estimator/param_estimates/m"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="r" containers="1">
   <Container>
    <DockSplitter sizes="0.333037;0.333927;0.333037" count="3" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-0.013618" top="0.057479"/>
       <limitY/>
       <curve color="#d62728" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BI/x"/>
       <curve color="#1ac938" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BI/y"/>
       <curve color="#ff7f0e" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BI/z"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-0.016604" top="0.029849"/>
       <limitY/>
       <curve color="#f14cc1" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BM/x"/>
       <curve color="#9467bd" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BM/y"/>
       <curve color="#17becf" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BM/z"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="-0.064949" top="0.044918"/>
       <limitY/>
       <curve color="#bcbd22" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BP/x"/>
       <curve color="#1f77b4" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BP/y"/>
       <curve color="#d62728" name="/quadrotor/geom_inertia_estimator/param_estimates/r_BP/z"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="rpm" containers="1">
   <Container>
    <DockSplitter sizes="0.250446;0.249554;0.250446;0.249554" count="4" orientation="-">
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="5325.200000" top="16866.800000"/>
       <limitY/>
       <curve color="#d62728" name="/quadrotor/rpm/rpm[0]"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="5070.000000" top="16746.000000"/>
       <limitY/>
       <curve color="#1ac938" name="/quadrotor/rpm/rpm[1]"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="7285.100000" top="16814.900000"/>
       <limitY/>
       <curve color="#ff7f0e" name="/quadrotor/rpm/rpm[2]"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot mode="TimeSeries" style="Lines" flip_x="false" flip_y="false">
       <range left="0.000000" right="25.138275" bottom="4939.400000" top="16892.600000"/>
       <limitY/>
       <curve color="#f14cc1" name="/quadrotor/rpm/rpm[3]"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="7"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <parameters time_axis="" delimiter="0"/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value="/quadrotor/geom_inertia_estimator/param_estimates;/quadrotor/imu;/quadrotor/pose;/quadrotor/rpm"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value="/quadrotor/geom_inertia_estimator/param_estimates;/quadrotor/pose;/quadrotor/imu;/quadrotor/rpm"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles>
  <fileInfo filename="result.bag" prefix="">
   <plugin ID="DataLoad ROS bags">
    <use_header_stamp value="false"/>
    <discard_large_arrays value="true"/>
    <max_array_size value="100"/>
    <boolean_strings_to_number value="true"/>
    <remove_suffix_from_strings value="true"/>
    <selected_topics value="/quadrotor/geom_inertia_estimator/param_estimates;/quadrotor/imu;/quadrotor/pose;/quadrotor/rpm"/>
   </plugin>
  </fileInfo>
 </previouslyLoaded_Datafiles>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

